### PR TITLE
Autopark level 13

### PR DESCRIPTION
**Pull request by AlexandarSviridenko**
Autopark level 13 done.

### CODE REVIEW

File: src/main/java/com/Autopark/infrastructure/config/impl/JavaConfig.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private final Scanner scanner;
+    private final Map<Class<?>, Class<?>> interfaceToImplementation;
+
+    @Override
+    public <T> Class<? extends T> getImplementation(Class<T> target) {
+        Set<Class<? extends T>> implementation = scanner.getSubtypesOf(target);
+
+        if (implementation.size() != 1) {
+            if (!isImplementationPresent(target)) {
+                throw new RuntimeException("target interface has 0 or more than 1 implementation");
```
Comment (AlexeyKatomachin) [inline src/main/java/com/Autopark/infrastructure/config/impl/JavaConfig.java:20]: Maybe better create own exception?

File: src/main/java/com/Autopark/ParserVehicleFromFile.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                return vehicle;
+            }
+        }
+        throw new NoSuchElementException();
+    }
+
+    private Startable createEngine(String[] params, int order) {
+        String engineStr = params[order];
+
+        if (engineStr.equalsIgnoreCase("Electrical")) {
```
Comment (AlexeyKatomachin) [inline src/main/java/com/Autopark/ParserVehicleFromFile.java]: maybe in this case better use switch?

File: src/main/java/com/Autopark/ParserVehicleFromFile.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        return params;
+    }
+
+    private VehicleType getTypeById(int typeId) {
+        for (VehicleType type : loadTypes()) {
+            if (type.getId() == typeId) {
+                return type;
+            }
+        }
+        throw new NoSuchElementException();
```
Comment (AlexeyKatomachin) [inline src/main/java/com/Autopark/ParserVehicleFromFile.java]: Maybe better create own exception?

Comment (AlexandarSviridenko) [reply src/main/java/com/Autopark/ParserVehicleFromFile.java]: Сorrected.

File: src/main/java/com/Autopark/infrastructure/config/impl/JavaConfig.java (hunk 1 above)
Comment (AlexandarSviridenko) [reply src/main/java/com/Autopark/infrastructure/config/impl/JavaConfig.java:20]: Maybe, but in a Task as needed use RuntimeException.

File: src/main/java/com/Autopark/ParserVehicleFromFile.java (hunk 2 above)
Comment (AlexandarSviridenko) [reply src/main/java/com/Autopark/ParserVehicleFromFile.java]: I agree. Code corrected.

**Review by AlexeyKatomachin — APPROVED** [review]

**Pull request merged**